akka-http-core/src/test/scala/akka/http/scaladsl/model/UriSpec.scala

```diff
@@ -721,11 +722,23 @@ class UriSpec extends WordSpec with Matchers {
       Uri("https://host:3030/").withPort(4450).effectivePort shouldEqual 4450
     }
 
+    "parse authority" in {
+      Uri.Authority.parse("localhost").toString shouldEqual "localhost"
+      Uri.Authority.parse("example.com:80").toString shouldEqual "example.com:80"
+      Uri.Authority.parse("user@host").toString shouldEqual "user@host"
+
+      val a = Uri.Authority.parse("user:p%40ssword@host")
+      a.userinfo.length shouldEqual "user:p%40ssword".length
```

Comment:
`a.userinfo shouldEqual "user:p@ssword"` would be correct.

Reply:
Hmm, I see so we SHOULD interpret it during parsing... I'll give it a shot